plugins/common/common.cpp

```diff
@@ -1298,11 +1312,15 @@ void AuiToolBar::OnTool( wxCommandEvent& event )
 		return;
 	}
 
-	wxObject* wxobject = (wxObject*) tb->FindTool( event.GetId() )->GetUserData();
-
-	if ( NULL != wxobject )
+	wxAuiToolBarItem* item = tb->FindTool(event.GetId());
+	if (item)
 	{
-		m_manager->SelectObject( wxobject );
+		wxObject* wxobject = (wxObject*) GetPtrData(item->GetUserData());
```

Comment:
Modern-C++: Please no c-style casts. I'm quite a fan of `auto`, especially if the type is already used explicit on the right side, so this could be changed to `auto* wxobject = static_cast<wxObject*>(GetPtrData(item->GetUserData()));`. However this isn't necessary anymore if the lookup map returns `wxObject*`.